File: app/src/main/java/konradrutkowski/com/tapapp/activities/DetailsActivity.java
package konradrutkowski.com.tapapp.activities;

import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;


import com.squareup.picasso.Picasso;

import konradrutkowski.com.tapapp.customviews.CustomScrollView;
import konradrutkowski.com.tapapp.places.model.Place;
import konradrutkowski.com.tapapp.R;

public class DetailsActivity extends AppCompatActivity {

    ActionBar actionBar;
    ColorDrawable newColor = null;
    boolean titleDisplay = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        setContentView(R.layout.details_fragment);
        Intent intent = this.getIntent();
        Bundle bundle = intent.getExtras();
        Place place = (Place) bundle.getSerializable("fsquareObj");
        actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
            actionBar.setElevation(0);
            newColor = new ColorDrawable(getResources().getColor(R.color.ColorFront));
            newColor.setAlpha(25);
            actionBar.setBackgroundDrawable(newColor);
            actionBar.setDisplayShowTitleEnabled(titleDisplay);
            if (place != null) {
                actionBar.setTitle(place.getName());
            }
        }


        CustomScrollView scrollView = (CustomScrollView) findViewById(R.id.detailsScrollView);
        scrollView.setOnScrollViewListener(new CustomScrollView.OnScrollViewListener() {

            @Override
            public void onScrollChanged(CustomScrollView v, int l, int t, int oldl, int oldt) {
                newColor.setAlpha(getAlphaforActionBar(v.getScrollY()));
                actionBar.setBackgroundDrawable(newColor);
                actionBar.setDisplayShowTitleEnabled(titleDisplay);
            }

            private int getAlphaforActionBar(int scrollY) {
                int minDist = 10, maxDist = 450;
                if (scrollY > maxDist) {
                    titleDisplay = true;
                    return 255;
                } else if (scrollY < minDist) {
                    titleDisplay = false;
                    return 10;
                } else {
                    titleDisplay = false;
                    int alpha = 0;
                    alpha = (int) ((255.0 / maxDist) * scrollY);
                    return alpha;
                }
            }
        });
        initView(place);

        ImageView imageView = (ImageView) findViewById(R.id.imageBig);
        if (place != null) {
            Picasso.with(this).load(place.getUrl()).into(imageView);
        }

        final FloatingActionButton addToFavButton = (FloatingActionButton) findViewById(R.id.addToFav);
        addToFavButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View arg0) {
                Toast.makeText(getApplication(), "TODO Favorite section ", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    protected void onPause() {
        super.onPause();
        overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_right);
    }

    private void initView(Place mPlace) {
        TextView name = (TextView) findViewById(R.id.placeNameText);
        TextView category = (TextView) findViewById(R.id.categoryText);
        TextView distance = (TextView) findViewById(R.id.distanceText);
        TextView city = (TextView) findViewById(R.id.cityText);
        TextView checkins = (TextView) findViewById(R.id.chekinsText);
        TextView address = (TextView) findViewById(R.id.addressText);
        //

        if (mPlace.getName() != null) {
            name.setText(mPlace.getName());
        } else {
            name.setText("Not available");
        }

        if (mPlace.getCategory() != null) {
            category.setText(mPlace.getCategory());
        } else {
            category.setText("Not available");
        }
        if (mPlace.getDistance() != null) {
            distance.setText(mPlace.getDistance() + " meters");
        } else {
            distance.setText("Not available");
        }
        if (mPlace.getCheckins() != null) {
            checkins.setText(mPlace.getCheckins());
        } else {
            checkins.setText("Not available");
        }
        if (mPlace.getCity() != null) {
            city.setText(mPlace.getCity());
        } else {
            city.setText("Not available");
        }
        if (mPlace.getAddress() != null) {
            address.setText(mPlace.getAddress());
        } else {
            address.setText("Not available");
        }
    }
}

File: app/src/main/java/konradrutkowski/com/tapapp/activities/DetailsActivity.kt
package konradrutkowski.com.tapapp.activities

import android.content.pm.ActivityInfo
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import konradrutkowski.com.tapapp.R
import konradrutkowski.com.tapapp.customviews.CustomScrollView
import konradrutkowski.com.tapapp.fsquare.Place

class DetailsActivity : AppCompatActivity() {

    internal var actionBar: ActionBar? = null
    internal var newColor: ColorDrawable? = null
    internal var titleDisplay = false

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        setContentView(R.layout.details_fragment)
        val intent = this.intent
        val bundle = intent.extras
        val mFSquarePlace = bundle!!.getSerializable("fsquareObj") as Place
        actionBar = supportActionBar
        if (actionBar != null) {
            actionBar!!.setDisplayHomeAsUpEnabled(true)
            actionBar!!.setDisplayShowHomeEnabled(true)
            actionBar!!.elevation = 0f
            newColor = ColorDrawable(resources.getColor(R.color.ColorFront))
            newColor!!.alpha = 25
            actionBar!!.setBackgroundDrawable(newColor)
            actionBar!!.setDisplayShowTitleEnabled(titleDisplay)


            actionBar!!.title = mFSquarePlace.name
        }


        val scrollView = findViewById<View>(R.id.detailsScrollView) as CustomScrollView
        scrollView.setOnScrollViewListener(object : CustomScrollView.OnScrollViewListener {

            override fun onScrollChanged(v: CustomScrollView, l: Int, t: Int, oldl: Int, oldt: Int) {
                newColor!!.alpha = getAlphaforActionBar(v.scrollY)
                actionBar!!.setBackgroundDrawable(newColor)
                actionBar!!.setDisplayShowTitleEnabled(titleDisplay)
            }

            private fun getAlphaforActionBar(scrollY: Int): Int {
                val minDist = 10
                val maxDist = 450
                if (scrollY > maxDist) {
                    titleDisplay = true
                    return 255
                } else if (scrollY < minDist) {
                    titleDisplay = false
                    return 10
                } else {
                    titleDisplay = false
                    var alpha: Int
                    alpha = (255.0 / maxDist * scrollY).toInt()
                    return alpha
                }
            }
        })
        initView(mFSquarePlace)

        val imageView = findViewById<View>(R.id.imageBig) as ImageView
        if (mFSquarePlace.url != null) {
            //  imageLoader.displayImage(mFSquarePlace.getUrl(), imageView, options);
        } else {
            imageView.visibility = View.GONE
        }

        val addToFavButton = findViewById<View>(R.id.addToFav) as FloatingActionButton
        addToFavButton.setOnClickListener { Toast.makeText(application, "TODO Favorite section ", Toast.LENGTH_SHORT).show() }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                this.finish()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    override fun onPause() {
        super.onPause()
        overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_right)
    }

    private fun initView(mPlace: Place) {
        val name = findViewById<View>(R.id.placeNameText) as TextView
        val category = findViewById<View>(R.id.categoryText) as TextView
        val distance = findViewById<View>(R.id.distanceText) as TextView
        val city = findViewById<View>(R.id.cityText) as TextView
        val checkins = findViewById<View>(R.id.chekinsText) as TextView
        val address = findViewById<View>(R.id.addressText) as TextView
        //

        if (mPlace.name != null) {
            name.text = mPlace.name
        } else {
            name.text = "Not available"
        }

        if (mPlace.category != null) {
            category.text = mPlace.category
        } else {
            category.text = "Not available"
        }
        if (mPlace.distance != null) {
            distance.text = mPlace.distance + " meters"
        } else {
            distance.text = "Not available"
        }
        if (mPlace.checkins != null) {
            checkins.text = mPlace.checkins
        } else {
            checkins.text = "Not available"
        }
        if (mPlace.city != null) {
            city.text = mPlace.city
        } else {
            city.text = "Not available"
        }
        if (mPlace.address != null) {
            address.text = mPlace.address
        } else {
            address.text = "Not available"
        }
    }
}
